price-estimator/src/amounts_at_price.rs

```diff
@@ -6,9 +6,15 @@ pub fn order_at_price_with_rounding_buffer(
     token_pair: TokenPair,
     limit_price: f64,
     pricegraph: &Pricegraph,
-    rounding_buffer: f64,
+    rounding_buffer: Option<f64>,
 ) -> Option<TransitiveOrder> {
     let order = pricegraph.order_for_limit_price(token_pair, limit_price)?;
+
+    let rounding_buffer = match rounding_buffer {
+        None => return Some(order),
+        Some(rounding_buffer) => rounding_buffer,
+    };
```

Comment:
This looks like some kind of weird type manipulation. I read, if there is a rounding buffer, then rounding buffer is itself unwrapped, otherwise return the order.

Perhaps it would be more clear if we simply use an if statement

Reply:
If I use an if then the whole rest of the function has to be part of a new block which I'd like to avoid. I wouldn't call this a weird manipulation either, just an early return in case rounding buffer is none.